apps/condo/domains/meter/utils/serverSchema/resolveHelpers.js

```diff
@@ -40,6 +43,16 @@ async function addClientInfoToResidentMeterReading (context, resolvedData) {
     resolvedData.clientEmail = user.email
 }
 
+async function connectContactToMeterReading (context, resolvedData, existingItem) {
+    const resolvedContact = get(resolvedData, 'contact', null)
+
+    if (!resolvedContact) {
+        const contact = await getOrCreateContactByClientData(context, resolvedData, existingItem)
+        resolvedData.contact = contact.id
```

Comment:
You are mutating `resolvedData` within some function. Would it be better to return the required value?

Reply:
Many functions in `resolveHelpers` work similarly in other domains

Reply:
🥲